test/unit_test/automl_weights_test.cc

```diff
@@ -325,3 +326,64 @@ BOOST_AUTO_TEST_CASE(automl_equal_no_automl_w_iterations)
   BOOST_CHECK(qcolcol_weights_vector == automl_champ_weights_vector);
   BOOST_CHECK(ctr1 == ctr2);
 }
+
+BOOST_AUTO_TEST_CASE(automl_equal_spin_off_model)
+{
+  callback_map test_hooks;
+
+  std::string vw_arg =
+      "--cb_explore_adf --quiet --epsilon 0.2 "
+      "--random_seed 5 --predict_only_model ";
+  std::string vw_automl_arg =
+      "--automl 4 --priority_type favor_popular_namespaces "
+      "--oracle_type one_diff --global_lease 10 ";
+  int seed = 10;
+  // a switch happens around ~1756
+  size_t num_iterations = 1700;
+
+  auto* vw_qcolcol = VW::initialize(vw_arg + "-b 17 -q ::");
```

Comment:
can we save the model here and then after calling vw::finish hash both and make sure its the same contents? - we would also have to delete the model file to be good citizens

Reply:
https://github.com/VowpalWabbit/vowpal_wabbit/pull/4279/files#diff-dbdeee57e850cde4835b15d683271658db36cb50cc9e288deaf1a6beff534a5fR5511

This test checks that the model files are identical